src/nodeViews/createReactNodeViewConstructor.tsx

```diff
@@ -215,9 +247,11 @@ export function createReactNodeViewConstructor(
     );
 
     const unregisterElement = registerElement(
+      findNearestRegistryKey(editorView, getPos()),
+      getPos,
       element,
       dom as HTMLElement,
-      Math.floor(Math.random() * 0xffffff).toString(16)
+      key
```

Comment:
Is there any reason to have the portal registry key and the React key be different?

Reply:
The registration key isn't unique; all of the children of a given node will register with the same key. But they're not actually different; the react key is the key that the children of _this node_ will register with. Does that make sense? Hoping I'm answering the right question haha.

Reply:
Ahh, right. Okay. And `key` _is_ the registry key of this node. Maybe naming changes can help. We have a key for ourselves, and the key of the nearest ancestor, yeah?

Are the arguments here getting busy? Should we move `createPortal` here and then register that result?

Reply:
I think the naming I'm referring to is that "registry key" led me to think one key was a registry key and one key was a rendering key, but both are the same "kind" of key, the difference is which element they refer to.

Reply:
Agreed, the naming is not super clear! I'll try to see what I can come up with (I'll try registering just the portal, too, that makes sense)